src/sdltiles.cpp

```diff
@@ -244,6 +256,52 @@ extern catacurses::window w_hit_animation; //this window overlays w_terrain whic
 //***********************************
 //Non-curses, Window functions      *
 //***********************************
+void generate_alt_rect_texture()
+{
+#if SDL_BYTEORDER == SDL_BIG_ENDIAN
+    static const Uint32 rmask = 0xff000000;
+    static const Uint32 gmask = 0x00ff0000;
+    static const Uint32 bmask = 0x0000ff00;
+    static const Uint32 amask = 0x000000ff;
+#else
+    static const Uint32 rmask = 0x000000ff;
+    static const Uint32 gmask = 0x0000ff00;
+    static const Uint32 bmask = 0x00ff0000;
+    static const Uint32 amask = 0xff000000;
+#endif
+
+    try {
+      SDL_Surface_Ptr alt_surf( SDL_CreateRGBSurface( 0, 1, 1, 32, rmask, gmask, bmask, amask ) );
+      SDL_FillRect( alt_surf.get(), NULL, SDL_MapRGB( alt_surf->format, 255, 255, 255 ) );
+      alt_rect_tex.reset( SDL_CreateTextureFromSurface( renderer.get(), alt_surf.get() ) );
+      alt_surf.reset();
+
+      // test to make sure color modulation is supported by renderer
+      if( SDL_SetTextureColorMod( alt_rect_tex.get(), 0, 0, 0 ) ){
+          alt_rect_tex_enabled = false;
+          alt_rect_tex.reset();
+          dbg( D_ERROR ) << "generate_alt_rect_texture color modulation test failed: " << SDL_GetError();
+      } else {
+          alt_rect_tex_enabled = true;
+      }
+    } catch( const std::exception &err ) {
```

Comment:
Why are you catching here? What of the above can throw? SDL2 is a C library, it does not throw a thing, you need to check the function call results to detect errors.

Reply:
Ok